addons/crm_iap_lead_enrich/models/crm_iap_lead_enrich.py

```diff
@@ -0,0 +1,120 @@
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+from odoo.addons.iap import jsonrpc,InsufficientCreditError
+
+DEFAULT_ENDPOINT = 'https://iap-services.odoo.com'
+
+class Lead(models.Model):
+    _inherit = 'crm.lead'
+
+    enriched_lead = fields.Boolean(string='Lead enriched based on email', default=False)
+
+    @api.multi
+    def lead_enrich_mail(self):
+        request = {}
+        for record in self:
+            if record.probability == 0:
+                continue
+            email_domain = record.partner_address_email or record.email_from
+            if email_domain:
+                request[record.id] = email_domain.split('@')[1]
+            else:
+                record.message_post_with_view('crm_iap_lead_enrich.lead_enrich_message_no_mail', subtype_id=self.env.ref('mail.mt_note').id)        
+        try:
+            response_clearbit = self._make_request(request)
+            self._enrich_leads_from_response(response_clearbit)
+        except InsufficientCreditError:
+            for record in self:
+                data = {
+                    'message_title': _("Lead enriched based on email address"),
+                    'url' : self.env['iap.account'].get_account_url(),
+                }
+                record.message_post_with_view('crm_iap_lead_enrich.lead_enrich_message_no_credit', values=data, subtype_id=self.env.ref('mail.mt_note').id)
+    
+    @api.model
+    def enrich_with_cron(self):
+        leads = self.search([('enriched_lead', '=', False),('reveal_id', '=', False)])
```

Comment:
Hmmm, make no sense, all old leads, leads in won stage, ... will be processed for nothing... We should change the condition to something more restrictive.

Reply:
who this be restrictive enough? 
leads = self.search([('enriched_lead', '=', False),('reveal_id', '=', False),('active','=',True),('won_status','=','pending')])

Reply:
When you search on records, active=True is implicit, you do not need to add this condition.
For the condition, you can have a look in the task :)